services/media/service/src/domains/common/utils/localization-utils.ts

```diff
@@ -0,0 +1,79 @@
+import { Dict } from '@axinom/mosaic-db-common';
+import {
+  DeleteLocalizationSourceEntityCommand,
+  LocalizationServiceMultiTenantMessagingSettings,
+  UpsertLocalizationSourceEntityCommand,
+  UpsertLocalizationSourceEntityFields,
+} from '@axinom/mosaic-messages';
+import { MosaicError } from '@axinom/mosaic-service-common';
+import { Table } from 'zapatos/schema';
+import { InternalErrors } from '../../../common';
+import { LocalizationMessageData } from '../models';
+
+export const getUpsertMessageData = (
+  serviceId: string,
+  entityType: string,
+  entityId: number,
+  fields: UpsertLocalizationSourceEntityFields,
+  entityTitle: string | undefined,
+  imageId: string | null | undefined,
+): LocalizationMessageData => {
+  const payload: UpsertLocalizationSourceEntityCommand = {
+    service_id: serviceId,
+    entity_type: entityType,
+    entity_id: entityId.toString(),
+    entity_title: entityTitle,
+    image_id: imageId,
+    fields,
+  };
+  return {
+    settings:
+      LocalizationServiceMultiTenantMessagingSettings.UpsertLocalizationSourceEntity,
+    payload,
+  };
+};
+
+export const getDeleteMessageData = (
+  serviceId: string,
+  entityType: string,
+  entityId: number,
+): LocalizationMessageData => {
+  const payload: DeleteLocalizationSourceEntityCommand = {
+    service_id: serviceId,
+    entity_type: entityType,
+    entity_id: entityId.toString(),
+  };
+  return {
+    settings:
+      LocalizationServiceMultiTenantMessagingSettings.DeleteLocalizationSourceEntity,
+    payload: payload,
+  };
+};
+
+export const assertLocalizationData: (
+  data: Dict<unknown> | undefined,
+  tableName: Table,
+) => asserts data is Dict<unknown> = (
+  data: Dict<unknown> | undefined,
+  tableName: Table,
+): asserts data is Dict<unknown> => {
+  if (!data) {
+    throw new MosaicError({
+      ...InternalErrors.ReplicaDataNotFound,
+      messageParams: [tableName],
+    });
+  }
+};
+
+export const assertLocalizationColumn = (
+  data: Dict<unknown>,
+  columnName: string,
+  tableName: Table,
+): void => {
+  if (!(columnName in data)) {
```

Comment:
Maybe also here to be really sure: `typeof obj !== 'object' || obj === null `

Reply:
The reply above is slightly related to this one.
If we stick to not generalizing this function, then there is no need to check
this, because `assertLocalizationColumn` and `assertLocalizationData` are meant
to be checked in combination with one another to be able to produce
understandable and errors that are matching the failure condition.